Feature/5 add extended transformations (#6)
AbsaOSS/spark-hats: README.md

```

### Add a column
The `nestedWithColumn` method allows adding new fields inside nested structures and arrays.

The addition of a column API is provided in two flavors: the basic and the extended API. The basic API is simpler to

> use, but expressions it expects cannot reference columns on parent array levels. Here is an example of the basic add

Reviewer: ```suggestion
use, but the expressions it expects can only reference columns inside the array and the root of the schema. Here is an example of the basic add
```

+---+---------------------------------------------------+
```

### Add column (extended)

> The extended API method `nestedWithColumnExtended` works similar to the basic one, but allows the caller to reference

Reviewer: ```suggestion
The extended API method `nestedWithColumnExtended` works similarly to the basic one but allows the caller to reference
```

other array elements, possibly on different levels of nesting. The way it allows this is a little tricky.
The second parameter is changed from being a column to a *function that returns a column*. Moreover, this function has
an argument which is a function itself, the `getField()` function. The `getField()` function can be used in the
transformation to reference other columns in the dataframe by their fully qualified name.

> Int the following example, a transformation adds a new field `my_array.c` to the dataframe by concatenating a root

Reviewer: ```suggestion
In the following example, a transformation adds a new field `my_array.c` to the dataframe by concatenating a root
```
